core/src/main/scala/com/permutive/refreshable/Refreshable.scala

```diff
@@ -251,17 +302,17 @@ object Refreshable {
       } yield ()).uncancelable
 
     private def refreshLoop(
-        initialA: A,
+        initialA: CachedValue[A],
         fa: F[A],
-        set: CachedValue[A] => F[Unit],
+        set: (CachedValue[A], CachedValue[A]) => F[Unit],
         cacheDuration: A => FiniteDuration,
         onRefreshFailure: PartialFunction[(Throwable, RetryDetails), F[Unit]],
         onNewValue: (A, FiniteDuration) => F[Unit],
         retryPolicy: A => RetryPolicy[F]
-    ): F[Unit] = {
-      def innerLoop(currentA: A): F[Unit] = {
+    ): F[Unit] =
+      FlatMap[F].tailRecM[A, Unit](initialA.value) { currentA =>
```

Comment:
I don't believe that this is necessary (assuming that any `F` with a `Temporal` instance is stacksafe) but it does no harm anyway